pandas/core/indexes/base.py

```diff
@@ -2983,11 +2983,7 @@ def _union(self, other: Index, sort):
         elif not other.is_unique:
             # other has duplicates
 
-            # error: Argument 1 to "union_with_duplicates" has incompatible type
-            # "Union[ExtensionArray, ndarray]"; expected "ndarray"
-            # error: Argument 2 to "union_with_duplicates" has incompatible type
-            # "Union[ExtensionArray, ndarray]"; expected "ndarray"
-            result = algos.union_with_duplicates(lvals, rvals)  # type: ignore[arg-type]
```

Comment:
i _think_ we should only get here with ndarray[object] for both.  if that can be confirmed, then can just do a cast.

Reply:
Unfortunately with categorical we get there with an EA

Reply:
so i think the right way to handle this will be to override ExtensionIndex._union (or maybe just CI._union)